force-app/main/default/classes/NKS_ConversationEntryController.cls

```diff
@@ -0,0 +1,272 @@
+            CRM_Type__c = 'Event',
+            CRM_Sent_date__c = msgSession.NKS_Authentication_Timestamp__c,
+            CRM_Read__c = false,
+            CRM_Message_Text__c = authText
+        );
+    }
+
+    private static Message__c createMessage(ConversationEntry entry, MessagingSession session, Thread__c thread) {
+        Message__c message = new Message__c();
+        message.CRM_Thread__c = thread.Id;
+        message.CRM_Message_Text__c = entry.messageText;
+        message.CRM_Sent_Date__c = DateTime.newInstance(entry.serverReceivedTimestamp);
+        message.CRM_Read__c = false;
+        message.CRM_Type__c = 'Message';
+
+        if ('Agent'.equals(entry.sender.role)) {
+            User navUser = navUserInfoByUserId.get(entry.sender.subject);
+            if (navUser != null) {
+                message.CRM_From_User__c = navUser.Id;
+                message.CRM_From_NAV_Unit__c = navUser.Department;
+            }
+        } else {
+            message.CRM_From_Contact__c = session.EndUserContactId;
+        }
+
+        return message;
+    }
+
+    private static void insertMessagesAndUpdatedThreadCloseDate(List<Message__c> messages, String threadId) {
+        try {
+            insert messages;
+            updateThreadCloseDate(threadId, System.now());
+        } catch (Exception e) {
+            logger.error('Error saving messages: ' + e.getMessage(), null, CRM_ApplicationDomain.Domain.NKS);
+        }
+    }
+
+    private static void splitChatbotMessagesFromConversation(
+        List<ConversationEntry> entries,
+        List<ConversationEntry> chatbotMessages,
+        List<ConversationEntry> userMessages
+    ) {
+        Boolean automaticGreetingFound = false;
+        for (ConversationEntry entry : entries) {
+            if (entry.messageText == null) {
+                continue;
+            }
+            entry.messageText = entry.messageText.unescapeHtml4(); // Connect API returns HTML-encoded strings -> decode for further processing
+
+            if (isWelcomeMessage(entry)) {
+                automaticGreetingFound = true;
+                entry.sender.role = 'Agent'; // Welcome message is incorrectly set as User message
+                userMessages.add(entry); // Add welcome message to user messages
+                continue;
+            }
+
+            if (automaticGreetingFound) {
+                userMessages.add(entry);
+            } else {
+                chatbotMessages.add(entry);
+            }
+        }
+    }
+
+    // TODO: Only check sender.role == System to avoid this failing if we change the welcome message in the future?
+    private static Boolean isWelcomeMessage(ConversationEntry entry) {
+        return entry.messageText != null &&
+            'System'.equalsIgnoreCase(entry.sender.role) &&
+            (entry.messageText.containsIgnoreCase('Welcome to chat with NAV') ||
+            entry.messageText.containsIgnoreCase('Velkommen til chat med NAV'));
+    }
```

Comment:
Kan denne triggere feil på en chat som har vært innom to veiledere? Altså første chat er avsluttet og så i andre chatten blir da første samtale, som inneholder denne teksten, en del av "Frida" loggen. Ville det ført til problemer, eller er ikke første melding sendt fra System kanskje?